BraintreeCore/BTAPIClient.m

```diff
@@ -297,16 +302,13 @@ - (void)fetchOrReturnRemoteConfiguration:(void (^)(BTConfiguration *, NSError *)
                     }
                 }
             }
-            
-            // Important: Unlock semaphore in all cases
-            dispatch_semaphore_signal(semaphore);
-        }];
-        
-        dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
 
-        dispatch_async(dispatch_get_main_queue(), ^{
-            completionBlock(configuration, fetchError);
-        });
+            dispatch_async(dispatch_get_main_queue(), ^{
+                completionBlock(configuration, fetchError);
+            });
+            dispatch_group_leave(group);
+        }];
+        dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
```

Comment:
I think this should be a small wait time. Maybe the next request gets through even though the first one isn't.

Reply:
Mind elaborating on this @kastiglione?

If I'm understanding this correctly, the network library itself should retry / wait for the maximum wait time and call the callback with fetchError. In that case, this operation will pop from the dispatch queue and the next one will miss the cache & re-request again.

Reply:
It's a question of how long is the network wait time? My suggestion is that this should control how long to wait before trying the a new queued fetch (if one exists).

Reply:
For a concrete example:

1. First request is made
2. Second request is queued

Currently, second request will wait for however long the network library waits (including maybe retries).

I'm suggesting this function can own how long to wait for, instead of letting the networking decide. If it the first one is slow for some reason, a second one might get through. Just a thought.

Reply:
Gotcha, yep it makes sense what you're suggesting.

It seems like timeout is 60 sec (both default and what they pass in).
https://github.com/braintree/braintree_ios/blob/9a192145eb8c5ae03ca7903a70773fcbeeb63d1c/BraintreePayPal/PayPalOneTouch/Configuration/PPOTConfiguration.m#L37

I think the handler should be rewritten in that case, since requests in flight would be racing on `configuration` setting, and the  callers would just get a completion with nil value (not a proper error).

There's a third option of passing the timeout to the network lib, that way you get both